src/Pages/RoomList/RoomList.js

```diff
@@ -0,0 +1,454 @@
+import FilterDropDown from "./FilterDropDown/FilterDropDown";
+import PriceFilterDropDown from "./PriceFilterDropDown/PriceFilterDropDown";
+import PageButtons from "./PageButtons/PageButtons";
+import EmptyState from "./EmptyState/EmptyState";
+import Footer from "../../Components/Footer/Footer";
+import styled from "styled-components";
+import { flexCenter, flexAlignCenter } from "../../styles/Theme";
+
+const API = "http://192.168.219.144:8000";
+const LIMIT = 15;
+const RoomList = () => {
+  const history = useHistory();
+  const [data, setData] = useState([]);
+  const [homes, setHomes] = useState([]);
+  const [bedFilter, setBedFilter] = useState([]);
+  const [filters, setFilters] = useState([]);
+  const [isRoomTypeFilterPop, setIsRoomTypeFilterPop] = useState(false);
+
+  const [isPriceFilterPop, setIsPriceFilterPop] = useState(false);
+  const [isPriceFiltered, setIsPriceFiltered] = useState(false);
+
+  const [isFilterModalPop, setIsFilterModalPop] = useState(false);
+
+  const [filteredResult, setFilteredResult] = useState([]);
+  const [prevFilterState, setPrevFilterState] = useState([]);
+
+  const [queryRoom, setQueryRoom] = useState("");
+  const [queryPrice, setQueryPrice] = useState("");
+  const [queryCondition, setQueryCondition] = useState("");
+  const queryStrings = [queryRoom, queryPrice, queryCondition]
+    .filter((query) => query)
+    .join("&");
+
+  const historyPush = () => {
+    history.push(`/roomlist?${queryStrings}`);
+  };
+
+  const fetchFilteredData = () => {
+    fetch(`${API}/homes?${queryStrings}`)
+      .then((res) => res.json())
+      .then((data) => {
+        data && setHomes(data.homes);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const handlePopUp = () => {
+    if (!isFilterModalPop) {
+      const newFilters = _.cloneDeep(filters);
+      setPrevFilterState(newFilters);
+    } else {
+      setFilters(prevFilterState);
+    }
+    setIsFilterModalPop(!isFilterModalPop);
+    document.body.style.overflow = `${isFilterModalPop ? "unset" : "hidden"}`;
+  };
+
+  // [필터 추가하기] -> [침실과 침대] 항목 수량 변경 기능
+  const handleModifyBtn = (id, type) => {
+    let modifiedListTemp = [];
+    let modifiedStatus = filters[0].filterChecked;
+    type === "plus" ? modifiedStatus[id]++ : modifiedStatus[id]--;
+    filters[0].filterItem.forEach((item, index) => {
+      modifiedListTemp.push({
+        name: item,
+        amount: modifiedStatus[index],
+      });
+    });
+    const filteredItemTemp = {
+      filterType: filters[0].filterType,
+      filteredItem: modifiedListTemp,
+    };
+    setBedFilter(filteredItemTemp);
+  };
+
+  // [필터 추가하기] -> [편의시설, 시설, 건물유형, 지역] 항목 체크 기능
+  const handleCheckBtn = (id, filterType) => {
+    const checkedStatus = filters.map((filter, index) => {
+      index &&
+        filter.filterType === filterType &&
+        (filter.filterChecked[id] = !filter.filterChecked[id]);
+      return filter;
+    });
+    setFilters(checkedStatus);
+  };
```

Comment:
```suggestion
  const handleCheckBtn = (id, filterType) => {
    const checkedStatus = filters.map((filter, index) => {
      const isTypeMatched = filter.filterType === filterType;
      
      if (!index || !isTypeMatched) return filter;
      if (isTypeMatched) return !filter.filterChecked[id];
    });

    setFilters(checkedStatus);
  };
```
- 기존 데이터의 값을 직접 수정하지 않기 위해(불변성) map, filter, reduce 메소드를 사용합니다! 
- 현재 작성된 방식은 기존의 데이터에 직접 접근하여 내용을 수정하고, 그 값을 return 해주고 있습니다. 
- 수정된 코드 확인해주시기 바랍니다. 가독성도 고려해서 위와 같이 수정해보는 것도 좋을 것 같아요 🙌

Reply:
그러네요. setState랑 뭔가 혼동해서 코드를 짰나봐요 코드 훨씬 깔끔하고 예뻐졌어요 너무 감사합니다ㅠㅠ